pkg/KubeArmorOperator/runtime/runtime.go

```diff
@@ -5,19 +5,37 @@ package runtime
 
 import (
 	"os"
+	"path/filepath"
 	"strings"
 
+	"github.com/kubearmor/KubeArmor/KubeArmor/log"
 	"github.com/kubearmor/KubeArmor/pkg/KubeArmorOperator/common"
 	"go.uber.org/zap"
 )
 
+func DetectNRI(pathPrefix, runtime string) (string, string) {
+	for _, path := range common.ContainerRuntimeSocketMap[runtime] {
+		if _, err := os.Stat(filepath.Clean(pathPrefix + path)); err == nil || os.IsPermission(err) {
+			if strings.Contains(path, "nri") {
+				return "nri", path
+			}
+			return runtime, path
+		} else {
+			log.Warnf("%s", err)
+		}
+	}
+	return "NA", "NA"
+}
+
 func DetectRuntimeViaMap(pathPrefix string, k8sRuntime string, log zap.SugaredLogger) (string, string) {
 	log.Infof("Checking for %s socket\n", k8sRuntime)
 	if k8sRuntime != "" {
 		for _, path := range common.ContainerRuntimeSocketMap[k8sRuntime] {
 			if _, err := os.Stat(pathPrefix + path); err == nil || os.IsPermission(err) {
-				if k8sRuntime == "docker" && strings.Contains(path, "containerd") {
-					return "containerd", path
+				if (k8sRuntime == "docker" && strings.Contains(path, "containerd")) || k8sRuntime == "containerd" {
```

Comment:
we have a fallback logic as well at #L47 in case if this if condition matched and failed to detect runtime info, it will not go to fallback. can we go through it again.